### ISSUE TITLE
Support for modern JS

### ISSUE THREAD

**Issue by AyrA** [sunjw/jstoolnpp#141]
Private class fields are currently not supported, and auto-formatting a JS document will break them.

Example:

```js
class ClassWithPrivate {
  #privateField;
  #privateFieldWithInitializer = 42;

  #privateMethod() {
    // …
  }

  static #privateStaticField;
  static #privateStaticFieldWithInitializer = 42;

  static #privateStaticMethod() {
    // …
  }
}
```

When formatting turns into:

```js
class ClassWithPrivate {
	 # privateField;
	 # privateFieldWithInitializer = 42;

	 # privateMethod() {
		// …
	}

	static # privateStaticField;
	static # privateStaticFieldWithInitializer = 42;

	static # privateStaticMethod() {
		// …
	}
}
```

The `#` is part of the name and no space is allowed between this symbol and the property name. The auto formatter also inserts a space before the symbol, even if it's the first symbol on the current line. This leads to a wrong level of indentation.

Details: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes/Private_class_fields

**Comment by sunjw** [comment]
I'll try to make it fixed in next release.

**Comment by sunjw** [comment]
Fix after 7b53d38c2051c8e3f628418a5750042c3452a243
